src/components/StoreRevokeBlock.js

```diff
@@ -1,80 +1,162 @@
 import React, { Component } from "react";
+/** @jsx jsx */
+import { css, jsx } from "@emotion/core";
 import PropTypes from "prop-types";
+import { connect } from "react-redux";
 import HashColor from "./UI/HashColor";
 import HashColorInput from "./UI/HashColorInput";
-import { OrangeButton } from "./UI/Button";
+import Modal from "./UI/Modal";
 import { isValidCertificateHash } from "../components/utils";
+import {
+  verifyCertificateValidity,
+  getCertificateValidity
+} from "../reducers/api";
+import Divider from "./UI/Divider";
+import CertificateDropZone from "./CertificateDropZone";
+import { getIsVerifying } from "../reducers/application";
+import {
+  getRevokeCertificateHash,
+  updateRevokeCertificateHash
+} from "../reducers/admin";
+
+const certificateDropzone = css`
+  width: 100%;
+  text-align: center;
+`;
 
 class StoreRevokeBlock extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      certificateHash: "",
-      certificateHashIsValid: true
+      certificate: {},
+      certificateHashIsValid: true,
+      isModalVisible: false
     };
-
     this.onHashChange = this.onHashChange.bind(this);
     this.onRevokeClick = this.onRevokeClick.bind(this);
+    this.handleCertificateSelected = this.handleCertificateSelected.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  getCertificateStatus(certValidator) {
+    const validaor = this.props.revokeCertificateValidity[certValidator];
```

Comment:
typo => validator

Reply:
I'm not sure to understand the structure used here 
- validator (so what's return from revokeCertificateValidity)
- status (so what's we build)

status looks like a duplication of redux store ?

Could you explain :)

Reply:
CertificateDropzone requires the structure of
```
{
      verified: true,
      verifying: false,
      error: ""
}
```

to be passed as props into the dropzone to show the relevant messages